src/msg/async/rdma/Infiniband.cc

```diff
@@ -629,7 +629,13 @@ Infiniband::CompletionQueue::~CompletionQueue()
 
 int Infiniband::CompletionQueue::init()
 {
-  cq = ibv_create_cq(infiniband.device->ctxt, queue_depth, this, channel->get_channel(), 0);
+  int comp_vector = 0;
+  if (cct->_conf->name.is_osd()) {
+    comp_vector = std::stoi(cct->_conf->name.get_id())
+                    % infiniband.device->ctxt->num_comp_vectors;
```

Comment:
this assumes that the osd ids sharing this device are evenly distributed in the range of "$id % $num_comp_vectors". but this is not necessarily true. what if `num_comp_vectors` is 62, and the distribution of osds deployed on this host is like

```
osd.62
osd.124
osd.186
osd.248
```

?

yeah, this is but a coined use case, but i think it should be able to explain the point: we are hardwiring the OSD to certain combination of user setting. which is not necessarily true.

so as @changchengx suggests, i think i'd be better to defer this decision to user. and this setting should be planned when creating/relocating an osd.